packages/providers/src.ts/base-provider.ts

```diff
@@ -830,28 +830,39 @@ export class BaseProvider extends Provider {
         // }
     }
 
+    /**
+     *  Get contract logs implementation, using the REST Api. 
+     *  It returns the logs array, or a default value []. 
+     *  Throws an exception, when the result size exceeds the given limit.
+     *
+     * @param filter The parameters to filter logs by.
+     */
     async getLogs(filter: Filter | Promise<Filter>): Promise<Array<Log>> {
         this._checkMirrorNode();
         const params = await resolveProperties({ filter: this._getFilter(filter) });
         const fromTimestampFilter = params.filter.fromTimestamp ? '&timestamp=gte%3A' + params.filter.fromTimestamp : "";
         const toTimestampFilter = params.filter.toTimestamp ? '&timestamp=lte%3A' + params.filter.toTimestamp : "";
-        const epContractsLogs = '/api/v1/contracts/' + params.filter.address + '/results/logs?limit=100';
+        const limit = 100;
+        const oversizeResponseLegth = limit + 1;
+        const epContractsLogs = '/api/v1/contracts/' + params.filter.address + '/results/logs?limit=' + oversizeResponseLegth;
         const requestUrl = this._mirrorNodeUrl + epContractsLogs + toTimestampFilter + fromTimestampFilter;
         try {
             let { data } = await axios.get(requestUrl);
             if (data) {
                 const mappedLogs = this.formatter.logsMapper(data.logs);
+                if (mappedLogs.length == oversizeResponseLegth) {
+                    logger.throwError(`query returned more than ${limit} results`, Logger.errors.CALL_EXCEPTION);
```

Comment:
`CALL_EXCEPTION` is used when contract calls revert. As this would be restriction from the mirror node, we should be returning `SERVER_ERROR`